internal/agent/device/dependency/dependency.go

```diff
@@ -389,6 +426,62 @@ func (m *prefetchManager) Cleanup() {
 	}
 }
 
+func (m *prefetchManager) cleanupPartialLayers(ctx context.Context) error {
+	if m.hasActivePulls() {
```

Comment:
While we only do a single pull at a time, I think Siri is right and if we did ever add multiple pullers we'd need to maintain holding the lock to ensure there are no active pulls. For now though I think a comment indicating that we don't allow multiple pulls in parallel could be useful.